plugin/src/test/java/io/jenkins/plugins/casc/misc/jmh/CascJmhBenchmarkState.java

```diff
@@ -0,0 +1,42 @@
+package io.jenkins.plugins.casc.misc.jmh;
+
+import io.jenkins.plugins.casc.ConfigurationAsCode;
+import io.jenkins.plugins.casc.ConfiguratorException;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nonnull;
+import jenkins.benchmark.jmh.JmhBenchmarkState;
+
+/**
+ * Use Configuration as Code to setup the Jenkins instance for JMH benchmark.
+ */
+public abstract class CascJmhBenchmarkState extends JmhBenchmarkState {
+    private static final Logger LOGGER = Logger.getLogger(CascJmhBenchmarkState.class.getName());
+
+    /**
+     * Location of the YAML file to be used for configuration-as-code
+     *
+     * @return String containing the location of YAML file in the classpath
+     */
+    @Nonnull
+    protected abstract String getResourcePath();
+
+    /**
+     * Setups the Jenkins instance using configuration as code
+     * available through the {@link CascJmhBenchmarkState#getResourcePath()}.
+     *
+     * @throws ConfiguratorException when unable to configure
+     */
+    @Override
+    public void setup() throws Exception {
```

Comment:
for `setup`, I suggest passing `this` so the resource loader will work for the current benchmark 😓
I don't know if there is a way to do it inside JUnit / benchmark framework

This how it is used here:

https://github.com/jenkinsci/configuration-as-code-plugin/blob/08b1a4867b0040a2de33a8a4d346df78caf659c2/plugin/src/test/java/io/jenkins/plugins/casc/misc/Util.java#L138-L160